src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java

```diff
@@ -70,9 +77,16 @@
 
 		FreestyleSolutionModel freestyleSolution = optionalFreestyleSolution.get();
 		freestyleSolution.setSolution(dto.getSolution());
-		return ControllerResult.of(this.freestyleSolutionRepository.save(freestyleSolution));
+		freestyleSolution = this.freestyleSolutionRepository.save(freestyleSolution);
+
+		return ControllerResult.of(new UpdateFreestyleSolutionResponseDTO(
+				freestyleSolution.getFreestyleSolutionId(),
+				freestyleSolution.getSolution()
+				)
+		);
 	}
 
+	@Transactional
```

Comment:
Why?

Reply:
because hibernates needs to know if i call a mehtod where a model is used
otherwise it throws an exception